crates/kitsune_p2p/kitsune_p2p/src/host_api.rs

```diff
@@ -0,0 +1,48 @@
+use must_future::MustBoxFuture;
+use std::sync::Arc;
+
+use kitsune_p2p_types::{bin_types::KitsuneSpace, dht_arc::DhtArcSet};
+
+use crate::event::{GetAgentInfoSignedEvt, MetricRecord};
+
+/// A boxed future result with dynamic error type
+pub type KitsuneHostResult<'a, T> =
+    MustBoxFuture<'a, Result<T, Box<dyn Send + Sync + std::error::Error>>>;
+
+/// The interface to be implemented by the host, which handles various requests
+/// for data
+pub trait KitsuneHost {
```

Comment:
I vote for putting some additional explicit bounds on this trait, it tends to move any compile errors inwards closer to where the actual coding issue is.

```suggestion
pub trait KitsuneHost: 'static + Send + Sync {
```